benchmark/sparse/linear_regression/mxnet_sparse_model.py

```diff
@@ -0,0 +1,47 @@
+"""
+Linear Regression model with sparse synthetic data for MXNet
+"""
+
+import time
+import mxnet as mx
+
+
+def run_benchmark(train_data, train_label, eval_data, eval_label, batch_size, epochs):
+    train_iter = mx.io.NDArrayIter(train_data, train_label, batch_size=batch_size, shuffle=True,
+                                   label_name='lin_reg_label', last_batch_handle='discard')
+    eval_iter = mx.io.NDArrayIter(eval_data, eval_label, batch_size=batch_size, shuffle=False,
+                                  label_name='lin_reg_label', last_batch_handle='discard')
+
+    X = mx.sym.Variable('data', stype='csr')
+    Y = mx.symbol.Variable('lin_reg_label', stype='csr')
+    fully_connected_layer = mx.sym.FullyConnected(data=X, name='fc1', num_hidden=10)
+    lro = mx.sym.LinearRegressionOutput(data=fully_connected_layer, label=Y, name="lro")
+
+    model = mx.mod.Module(
+        symbol=lro,
+        data_names=['data'],
+        label_names=['lin_reg_label']
+    )
+
+    start = time.time()
+
+    model.fit(train_iter, eval_iter,
+              optimizer_params={'learning_rate': 0.1, 'momentum': 0.9},
+              num_epoch=epochs,
+              eval_metric='mse',
+              batch_end_callback=mx.callback.Speedometer(batch_size))
+
+    print("MXNet Sparse Benchmark Results")
+    print("Batch Size")
+    print(batch_size)
+    print('Total Time')
+    print(time.time() - start)
+
+    metric = mx.metric.MSE()
+    mse = model.score(eval_iter, metric)
+    print("Achieved {0:.6f} validation MSE".format(mse[0][1]))
+    assert model.score(eval_iter, metric)[0][1] < 0.01001, "Achieved MSE (%f) is larger than expected (0.01001)" % \
```

Comment:
nit: why assert in benchmark script? 
Same comment for above tf script.

Reply:
Also, we are generating random data, so accuracy/loss doesn't add any value here?

Reply:
For first comment - I followed the style of the benchmarking script for linear regression model written in MXNet - https://mxnet.incubator.apache.org/tutorials/python/linear-regression.html

As we have a pre-defined test and train set, even if its synthetic we can loss/accuracy can still help us with understanding how the model is performing.

Reply:
Didn't track memory usage as we just wanted to check speed of mxnet vs keras-tf on sparse tensors.

Reply:
Agree that accuracy/loss has no value for synthetic data, even if the acc increase, it does not mean it's performing better. 
You can extend that in future if you want to include real dataset. and only check acc/loss on real dataset.

Reply:
About memory consumption , it's a good value add as sparse tensors save a lot of memory, and it's very easy to report. Just run on a GPU instance and report the output of `nvidia-smi`

Reply:
Yes thats a good point - I will update the memory usage in another PR